integration-cli/docker_api_containers_test.go

```diff
@@ -1616,3 +1616,10 @@ func (s *DockerSuite) TestPostContainersCreateWithOomScoreAdjInvalidRange(c *che
 		c.Fatalf("Expected output to contain %q, got %q", expected, string(b))
 	}
 }
+
+// test case for #22210 where an emtpy container name caused panic.
+func (s *DockerSuite) TestContainerApiDeleteWithEmptyName(c *check.C) {
+	status, _, err := sockRequest("DELETE", "/containers/", nil)
+	c.Assert(err, checker.IsNil)
+	c.Assert(status, checker.Equals, http.StatusNotFound)
```

Comment:
Hi @cpuguy83 `GetContainer(prefixOrName string)` will call `GetByName()` and only check if the name is nil or not:
https://github.com/docker/docker/blob/v1.11.0/daemon/daemon.go#L148-L152
If there is any error (e.g., empty string) it will move on to check
```
daemon.idIndex.Get(prefixOrName)
```
So the error code will still be 404.

Maybe the check of empty string should be placed inside `GetContainer(prefixOrName string)`?